### PR TITLE
Migrate and complete test suites

### PR DESCRIPTION
**Pull request by Hywan**
### Progression

- [x] `Bucket`,
- [x] `Composite`,
- [x] `Context`,
- [x] `Exception`,
- [x] `Filter`:
  - [x] `Basic`,
  - [x] `Exception`,
  - [x] `Filter`,
  - [x] `LateComputed`,
- [x] `IStream`:
  - [x] `Bufferable`,
  - [x] `In`,
  - [x] `Lockable`,
  - [x] `Out`,
  - [x] `Pathable`,
  - [x] `Pointable`,
  - [x] `Statable`,
  - [x] `Stream`,
  - [x] `Structural`,
  - [x] `Touchable`,
- [x] `Stream`,
- [x] `Wrapper`:
    - [x] `Exception`,
    - [x] `IWrapper`:
        - [x] `File`,
        - [x] `IWrapper`,
        - [x] `Stream`,
    - [x] `Wrapper`.

### CODE REVIEW

**Comment by Hywan** [comment]
One last class to test!

**Comment by Hywan** [comment]
Hello @hoaproject/hoackers,

So far, 2 bugs have been found and fixed. The last one involves your opinion! With `Hoa\Stream`, we are able to open, close and re-open a stream with the same instance. This is not dangerous, mostly safe (if the resource handler in PHP does not have any bugs). However, we got a bug here. If the stream buffer size were changed, when closing and re-opening, it was not reset. So we have a choice to do: Either reset the buffer size to the default size, or re-apply the change of the buffer size. Currently, I have decided to reset the buffer size (which is an attribute of the object, reflecting the current internal stream buffer size). From my point of view, this is more logic. Check this commit https://github.com/hoaproject/Stream/pull/21/commits/1cae8b54fe16c23bea019b5d6bc2d6a219da4fd4.

**Comment by Hywan** [comment]
```
Success (25 test suites, 125/125 test cases, 0 void test case, 0 skipped test case, 1466 assertions)!
```

\o/
And it's over.

File: Wrapper/Wrapper.php [hunk 1]
```diff
@@ -88,7 +89,7 @@ public static function register($protocol, $className, $flags = 0)
      */
     public static function unregister($protocol)
     {
-        return stream_wrapper_unregister($protocol);
+        return @stream_wrapper_unregister($protocol);
```
Comment (vonglasow) [inline Wrapper/Wrapper.php]: Why use here the silent operator ?

File: Wrapper/Wrapper.php [hunk 2]
```diff
@@ -99,7 +100,7 @@ public static function unregister($protocol)
      */
     public static function restore($protocol)
     {
-        return stream_wrapper_restore($protocol);
+        return @stream_wrapper_restore($protocol);
```
Comment (vonglasow) [inline Wrapper/Wrapper.php]: Why use here the silent operator ?

**Comment by vonglasow** [comment]
:+1:

File: Wrapper/Wrapper.php (hunk 1 above)
Comment (Hywan) [inline Wrapper/Wrapper.php]: This function already returns `false` if `$protocol` does not exist, but an error is triggered. This is not the expected behavior for our API. I will add an inline explanation.

File: Wrapper/Wrapper.php (hunk 2 above)
Comment (Hywan) [inline Wrapper/Wrapper.php]: Same as above https://github.com/hoaproject/Stream/pull/21/files/415b22279215e16c9142b585b83ce298b6dd8e84#r100768225.

**Comment by Hywan** [comment]
Thanks for the review!

**Pull request merged**